### ISSUE TITLE
Compatibility Suggestions

### ISSUE THREAD

**Issue by Ultimushadow** [RealMegaMinds/ClickOpener#10]
Hello, I was wondering if you could look into the following blocks from their respective mods:
- Supplementaries Safe and Sack items (they function similar to shulker boxes)
- Supplementaries Squared Colored Sack items (just the Supplementaries Sack in different colors)
- Fletching Table when paired with AdditionZ (adds a crafting GUI that does not store items)
- Trading Post from Trading Post (allows trading with all villagers within it's config'd radius)

By the way, could you tell me how you add this compat settings with these other mods? Are you just looking for an item to click on and the gui event it calls? Is there a way to more easily add modded guis into this via a config or is it complicated enough that it should just be hardcoded? Thanks!

**Comment by Hoid2** [comment]
I can look into adding those. I think when I was looking at it before there wasn't an easy way to add compatibility without doing it through code, but I can look again.

**Comment by Ultimushadow** [comment]
Yeah I figured it was probably a headache to try to config that but it might help avoid having to hardcode every time there's another mod with another block to access
I also noticed you only work with blocks, I was looking into the Item Filter from Interactive which only exists as an item, is that kind of inventory harder to access or is it just because the default compatibilities are all blocks?

**Comment by Hoid2** [comment]
I figured that items were easy enough to just be used normally but I can add that functionality too.

**Comment by Hoid2** [comment]
@Ultimushadow 
Version [3.0.0](https://modrinth.com/mod/clickopener/version/3.0.0-alpha+1.20.1) adds compatibility for the fletching table and trading post. For blocks without explicit support it fakes the player using the blocks, so lots of other blocks should work as well.

Items are next on todo

**Comment by Ultimushadow** [comment]
Perfect, I'll test out some more modded blocks soon and see if there are any bugs to iron out!

**Comment by Ultimushadow** [comment]
Alright, here are a couple blocks I've found don't work with the new system:

- Sack and Safe from mod Supplementaries
- Cartography Table when installed alongside DungeonZ

The first two don't give any error when clicked in the log, the last one does though because DungeonZ adds a right click with a Dungeon Compass. Honestly I'm more confused with the first two, the Safe is understandable maybe because of the use of a key item but the Sack is just a 9 slot Shulker Box. Do you have a check to see if the block uses an inventory to prevent issues? I tried various other blocks that I know shouldn't work and they all had inventory. 

Here are items that DO work in case it matters:

- Crafting stations from Chipped
- Diamond Grindstone from Spelunkery (just the menu obviously)
- Trading Post from Trading Post
- Fletching Table when alongside AdditionZ
- Runic Enchanting/Scribing Table (though like the enchanting table they aren't very useful when in inventory)

**Comment by Hoid2** [comment]
The enchanting table actually should change depending on the number of bookshelves around you. The scribing table should as well and is a bug if it doesn't. 
For blocks that don't cause errors but still don't work, there should be a message in the debug.log saying they failed to open.

By implementing the default handler I passed all decision making about what blocks are allowed to open to the mod users. It is easier for mod users to not add blocks to the config than it is for me to disallow each individual block that shouldn't be used. However, I did add a warning to the Modrinth page about the possibility of voiding contents.

Try this version https://modrinth.com/mod/clickopener/version/3.0.1+1.20.1. It adds compatibility for the sack from Supplementaries. The safe is still a work in progress.
Also the cartography table should no longer give errors. It still won't work currently because it is the dungeon compass that actually opens the screen not the table. This kind of functionality has been added to my todo list.

**Comment by Ultimushadow** [comment]
You know what, I actually never tested it because I assumed it just wouldn't work, it does work, thanks for the heads up!
I tested the Sack, the Cartography Table, Enchantment Table, Scribing Table, and the others from before to confirm that they're all working as intended!
Also as I thought, the Cartography Table does not work only when the player is holding a Dungeon Compass, no screen is open and no logs. I imagine that's because the Compass is creating the screen like you mentioned, and interfering with the Cartography Table in a way you can't avoid when it's in the player's hand? I tested the Diamond Grindstone from Spelunkery with a Rough gemstone (in world interaction happens on right click, polishing the gem) and it works in the inventory even with the gemstone is in the main hand, so I assume the gui from the Compass is definitely to blame for this behavior

**Comment by Hoid2** [comment]
Version [4.0.0](https://modrinth.com/mod/clickopener/version/4.0.0-alpha+1.20.1) should work for most things now. It includes support for items so the dungeon compass works, but I didn't test the Item Filter from Interactive. 
Supplementaries sack is no longer working because the author started doing client side stuff with it.
